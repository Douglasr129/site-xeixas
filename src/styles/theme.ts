/* eslint-disable quotes */
export const theme = {
	colors: {
		primaryColor: '#0a1128',
		secondaryColor: '#dc143c',
		white: '#FFFFFF',
		mediumGray: '#DDDDDD',
		darkColor: '#2d2d2d',
	},
	font: {
		family: {
			default: "'Open Sans', sans-serif",
			secundary: "'Montserrat',sans-serif",
		},
		sizes: {
			xsmall: '0.8rem',
			small: '1.6rem',
			medium: '2.4rem',
			large: '3.2rem',
			xlarge: '4.0rem',
			xxlarge: '4.8rem',
			huge: '5.6rem',
			xhuge: '6.4rem',
		},
	},
	media: {
		lteMedium: '(max-width: 768px)',
	},
	spacing: {
		xsmall: '0.8rem',
		small: '1.6rem',
		medium: '2.4rem',
		large: '3.2rem',
		xlarge: '4.0rem',
		xxlarge: '4.8rem',
		huge: '5.6rem',
		xhuge: '6.4rem',
	},
};
