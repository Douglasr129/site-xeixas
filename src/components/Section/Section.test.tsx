import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { Section } from '.';

describe('<Section />', () => {
	it('should render a Section', () => {
		renderTheme(
			<Section sectionId={'1'}>
				<h1>Children</h1>
			</Section>,
		);
		expect(
			screen.getByRole('heading', { name: 'Children' }),
		).toBeInTheDocument();
	});
	it('should render a Backgrout image', () => {
		const { container } = renderTheme(
			<Section background={'https://images/bg.jpg'} sectionId={'1'}>
				Children
			</Section>,
		);
		expect(container.firstChild).toHaveStyleRule(
			'background-image',
			'url(https://images/bg.jpg)',
		);
	});
	it('should render a Backgrout image', () => {
		const { container } = renderTheme(
			<Section background={'https://images/bg.jpg'} sectionId={'1'}>
				Children
			</Section>,
		);
		expect(container.firstChild).toMatchSnapshot();
	});
});
