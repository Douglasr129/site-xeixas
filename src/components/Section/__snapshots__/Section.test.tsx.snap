// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> should render a Backgrout image 1`] = `
.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background-image: url(https://images/bg.jpg);
  background-repeat: no-repeat;
  background-size: 100vw;
  background-position: center;
  margin: 0;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

<div
  class="c0"
  id="1"
>
  <div
    class="c1"
  >
    Children
  </div>
</div>
`;
